src/detect-engine.c

```diff
@@ -1970,6 +1970,10 @@ static int DetectEngineInspectRulePayloadMatches(
             if (!(s->flags & SIG_FLAG_REQUIRE_PACKET) && (p->flags & PKT_STREAM_ADD)) {
                 return false;
             }
+            if (s->flags & SIG_FLAG_REQUIRE_STREAM_ONLY) {
+                SCLogDebug("SIG_FLAG_REQUIRE_STREAM_ONLY, so no match");
+                return false;
```

Comment:
The function is defined as returning an `int` not `bool`. Suggest reconciling the types.

Reply:
Wondering if we could have a separate effort to address this and the other cases we see here, taking into account how these interact and spread into other parts of the engine...

Reply:
Due to how this is registered in the inspection API the func will stay `int`, but will update the return values to meet the API